### PR TITLE
Add libvncserver/0.9.14

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **libvncserver/0.9.14**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 glib uses 'libffi/3.4.4' and I have recently re-generated glib libraries. Why the CI is failing?
https://github.com/conan-io/conan-center-index/pull/20522

File: recipes/libvncserver/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        self.cpp_info.set_property("cmake_target_name", "libvncserver::libvncserver")
+        #self.cpp_info.libs = ["libvncserver", "libvncclient"]
+        self.cpp_info.libs = ['vncserver']
+        #self.cpp_info.libs = collect_libs(self)
+        #print(self.cpp_info.libs)
+
+        self.cpp_info.components["libvncclient"].set_property("cmake_target_name", "LibVncClient")
+        self.cpp_info.components["libvncclient"].libs = ['vncclient']
+        self.cpp_info.components["libvncclient"].set_property("cmake_target_name", "libvncserver::libvncclient")
+
```
Comment (uilianries) [inline recipes/libvncserver/all/conanfile.py:161]:
```suggestion
        self.cpp_info.components["server"].set_property("pkg_config_name", "libvncserver")
        self.cpp_info.components["client"].set_property("pkg_config_name", "libvncclient")

```
It generates pc files too: https://github.com/LibVNC/libvncserver/blob/LibVNCServer-0.9.14/CMakeLists.txt#L808

**Review by uilianries — COMMENTED** [review]
@AndreyMlashkin Thank you for this PR, I see this project is a real challenge, it requires tons of requirements and many of them are in middle of version conflict problem. It will require extra PRs to fix it 😞 

Some important conflict that are blocking this PR:

- 'gtk/4.7.0' requires 'libxml2/2.9.14' while 'xkbcommon/1.4.1' 
- 'glib/2.73.0' requires 'libffi/3.4.3' while 'wayland/1.22.0' requires 'libffi/3.4.4'.
- 'pango/1.50.7' requires 'freetype/2.12.1' while 'cairo/1.17.4' requires 'freetype/2.13.0'.
- 'pango/1.50.7' requires 'fontconfig/2.13.93' while 'cairo/1.17.4' requires 'fontconfig/2.14.2'.
- 'libsystemd/253.10' requires 'libmount/2.39' while 'glib/2.73.0' requires 'libmount/2.36.2'.
- 'gnutls/3.7.8' requires 'zstd/1.5.4' while 'libsystemd/253.10' requires 'zstd/1.5.5'.
-  'gnutls/3.7.8' requires 'brotli/1.0.9' while 'freetype/2.13.0' requires 'brotli/1.1.0'.

Note that you added some version ranges that are not allowed in CCI for now, and may cause more missing packages in case of using.

**Comment by AndreyMlashkin** [comment]
@uilianries , what's wrong about glib?
https://github.com/conan-io/conan-center-index/blob/9d98cdf1ec5720e93f6847f5c0d364e1b2442b43/recipes/glib/all/conanfile.py#L68


it uses libffi/3.4.4 for long time already and recently I have bumped other dependencies there

**Comment by AndreyMlashkin** [comment]
@uilianries and in the test_package - from where
#include <libvncserver/auth.h>
should come?

**Comment by AndreyMlashkin** [comment]
'glib/2.73.0' requires 'libffi/3.4.3' while 'wayland/1.22.0' requires 'libffi/3.4.4'.

Why is it keep complaining about glib, when I updated it 5 days ago?
https://github.com/conan-io/conan-center-index/pull/20522#issuecomment-1758975550

**Comment by AndreyMlashkin** [comment]
Strange... that lzo lib is included... I've tried to compile it with transitive_headers, but it does not help...
Any help from the community is appreciated

**Comment by AndreyMlashkin** [comment]
blocked by https://github.com/conan-io/conan-center-index/pull/20889

**Comment by AndreyMlashkin** [comment]
Okey, now I have the same problem as I have locally. Could please someone help with test package?
@uilianries

**Comment by AndreyMlashkin** [comment]
@uilianries , could you please generate binaries for
ERROR: Missing binary: ffmpeg/6.0:13846674305bc722baac7dc421c0f757ce8a108b
ERROR: Missing binary: vulkan-loader/1.3.239.0:c4cdea13add24a6aa62d7cdaf48a97d58dd8ecbd

**Comment by AndreyMlashkin** [comment]
@uilianries 
a question: when I use this recipe for conan v1 it requires a different name for find_package - in lowercase. Why?

**Comment by franramirez688** [comment]
> a question: when I use this recipe for conan v1 it requires a different name for find_package - in lowercase. Why?

Do you mean using legacy cmake generators? If you do that, legacy generators require the legacy attributes https://docs.conan.io/1/migrating_to_2.0/properties.html#migrating-legacy-cpp-info-attributes-to-set-property

**Comment by AndreyMlashkin** [comment]
@franramirez688 could you maybe generate binaries in the meantime?

> could you please generate binaries for
> ERROR: Missing binary: ffmpeg/6.0:13846674305bc722baac7dc421c0f757ce8a108b
> ERROR: Missing binary: vulkan-loader/1.3.239.0:c4cdea13add24a6aa62d7cdaf48a97d58dd8ecbd

**Comment by AndreyMlashkin** [comment]
@franramirez688 ping

**Comment by franramirez688** [comment]
> @franramirez688 ping

Looking into it. We rebuilt those binaries, but I'm afraid there is something else.

**Comment by AndreyMlashkin** [comment]
I have added v1 package to demonstrate my problem.
@franramirez688 @uilianries @SSE4
please help. There is naming inconsistency between conan v1 and conan v2 for find_package generator

**Comment by AndreyMlashkin** [comment]
With test_v1, it reproduces the problem I have with my conan v1 build.
Any help is appreciated.

**Comment by AndreyMlashkin** [comment]
> Looking into it. We rebuilt those binaries, but I'm afraid there is something else.

It should be something else, because I also don't see those libraries

**Comment by AndreyMlashkin** [comment]
@franramirez688 @uilianries 
now only ffmpeg binary is missing:
ERROR: Missing prebuilt package for 'ffmpeg/6.0'

If you have any clue why windows build fails, any help apreciated.
If not, I would throw exception for windows builds as first iteration of recipe implementation.

**Comment by AndreyMlashkin** [comment]
depends on https://github.com/conan-io/conan-center-index/pull/21088

**Comment by AndreyMlashkin** [comment]
depends https://github.com/conan-io/conan-center-index/pull/21088

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 
unfortunatelly, I can't reproduce this error in my conan v2 docker.
It builds just fine with Conan version 2.0.13

```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
[options]
*/*:shared=True

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
```

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin The correct Docker image to be used in this case should be `conanio/gcc11-ubuntu16.04:2.0.12`, the same used by the CI service.

Yes, I am able to reproduce with this one. Still no idea, why does it happening

**Comment by uilianries** [comment]
@AndreyMlashkin I checked the CMakeLists.txt and is a bit more trick:

- It uses variables for several dependencies (e.g. OPENSSL_LIBRARIES), instead of CMake targets. So you need to check if those variables are really aligned with those provided by CMakeDeps
- It uses find_package but as optional, in case some package does not make, it will silently fail to find the requirement
- It passes several defines (e.g. LIBVNCSERVER_HAVE_LIBJPEG), including for public headers. You will need to check it, than define transitive_headers/transitive_libs (if needed). Please, don't adding a comment where the transitive header is needed (e.g # INFO: rfbclient.h includes <sasl/sasl.h>). Plus, you need to export those defines in cpp_info as they are part of public headers too
- You need to add `self.cpp_info.components[<name>].requires` according to each component (client/server)

**Pull request closed without merging**